From 945e7b0d340dfe7f8297413e99b5d2db836c3bee Mon Sep 17 00:00:00 2001
From: Roqqit <roqqit@proton.me>
Date: Tue, 23 Sep 2025 18:05:30 +0000
Subject: [PATCH] Add more coverage to compress_tests

(cherry picked from commit 839f22407232fca4134cc331fce93dac4a8177ac)

Maintenance rationale (2026-10-14):
Exercise block-index compression round trips over the Dogecoin header/version representation, including the lineage-specific auxiliary proof state.

diff --git a/src/test/compress_tests.cpp b/src/test/compress_tests.cpp
index b9bdd89639e5b64e652a3969096e738cf2577598..65e1cbba0b664e29e91816c82d65986f2b2dce47 100644
--- a/src/test/compress_tests.cpp
+++ b/src/test/compress_tests.cpp
@@ -23,6 +23,9 @@
 // amounts 50 .. 21000000
 #define NUM_MULTIPLES_50BCH 420000
 
+// amounts 10000 .. 10000000000
+#define NUM_MULTIPLES_10000BCH 1000000
+
 BOOST_FIXTURE_TEST_SUITE(compress_tests, BasicTestingSetup)
 
 static bool TestEncode(Amount in) {
@@ -44,6 +47,10 @@ BOOST_AUTO_TEST_CASE(compress_amounts) {
     BOOST_CHECK(TestPair(COIN, 0x9));
     BOOST_CHECK(TestPair(50 * COIN, 0x32));
     BOOST_CHECK(TestPair(21000000 * COIN, 0x1406f40));
+    BOOST_CHECK(TestPair(10000000000 * COIN, 0x2540be400));
+    BOOST_CHECK(TestPair(MAX_MONEY, 0x2540be400));
+    BOOST_CHECK(TestPair(10000000001 * COIN, 0x14F46B0409));
+    BOOST_CHECK(TestPair(MAX_MONEY + COIN, 0x14F46B0409));
 
     for (int64_t i = 1; i <= NUM_MULTIPLES_UNIT; i++) {
         BOOST_CHECK(TestEncode(i * SATOSHI));
@@ -61,6 +68,10 @@ BOOST_AUTO_TEST_CASE(compress_amounts) {
         BOOST_CHECK(TestEncode(i * 50 * COIN));
     }
 
+    for (int64_t i = 1; i <= NUM_MULTIPLES_10000BCH; i++) {
+        BOOST_CHECK(TestEncode(i * 10000 * COIN));
+    }
+
     for (int64_t i = 0; i < 100000; i++) {
         BOOST_CHECK(TestDecode(i));
     }
